### PR TITLE
[#92949] Deactivate relay on auto logout

### PR DESCRIPTION
**Pull request by jdzak**
https://pm.tablexi.com/issues/92949

### CODE REVIEW

**Comment by jhanggi** [comment]
We should add a test that makes sure that it doesn't log out the user if there's a currently running reservation. I think that's handled inside `can_switch_instrument_off?`, but it would be nice to have that check.

**Comment by jdzak** [comment]
@jhanggi I added an spec to check a running reservation isn't killed by auto-logout. Was that what you were asking for?

**Comment by jdzak** [comment]
jhanggi said:
>There need to be two reservations. One that is going to get completed and moved to the problem queue > (I’m not it gets to the problem queue in this PR, but it might in one of your other ones), and another that > is currently running. We need to make sure that calling `complete_reservation` on the former, moves it > to completion, but doesn’t turn off the relay.

**Comment by jdzak** [comment]
@jhanggi Comments addressed.

File: app/support/auto_logout.rb [hunk 1]
```diff
@@ -23,6 +23,8 @@ def complete_status
   end
 
   def complete_reservation(od)
+    deactivate_relay = od.reservation.can_switch_instrument_off? && od.reservation.other_reservations_using_relay.count == 0
```
Comment (jhanggi) [inline app/support/auto_logout.rb]: I think this might be a good candidate to extract into a private method.

Comment (jhanggi) [inline app/support/auto_logout.rb]: I was initially thinking take both halves of the condition, but now I'm thinking just the `Reservation#other_reservation_using_relay?` into it's own method.

**Comment by jhanggi** [comment]
Just one minor comment. Otherwise, looks good.

File: app/support/auto_logout.rb (hunk 1 above)
Comment (jdzak) [inline app/support/auto_logout.rb]: Sure.

**Comment by jdzak** [comment]
@jhanggi Comments addressed.

**Comment by jdzak** [comment]
One second.. need to fix some failing specs.

**Comment by jdzak** [comment]
All good now.

**Comment by jhanggi** [comment]
After the previous merge, this is now in conflict in `app/support/auto_logout.rb`. Can you clean up that merge.

**Comment by jdzak** [comment]
Fixed the merge conflict. Also, I tweaked the logic in `ReservationSupport#other_reservation_using_relay?` to match the method name.

File: app/support/auto_logout.rb [hunk 2]
```diff
@@ -23,6 +23,8 @@ def complete_status
   end
 
   def complete_reservation(od)
+    deactivate_relay = !od.reservation.other_reservation_using_relay?
```
Comment (jhanggi) [inline app/support/auto_logout.rb]: What would you think of
```
relay = od.reservation.product.relay
relay.deactivate unless od.reservation.other_reservation_using_relay?
```
I think that reads pretty well.

**Comment by jhanggi** [comment]
Just one minor comment. Otherwise, looks good.

File: app/support/auto_logout.rb (hunk 2 above)
Comment (jdzak) [inline app/support/auto_logout.rb]: Yea, sounds better. I'll switch it.

**Comment by jdzak** [comment]
Ready for review.

**Pull request closed without merging**